Std/Data/List/Lemmas.lean

```diff
@@ -2066,8 +2070,8 @@ theorem forIn_eq_bindList [Monad m] [LawfulMonad m]
 /-! ### diff -/
 
 section Diff
--- TODO: theorems about `BEq`
```

Comment:
This doesn't eliminate the TODO, which is about having theorems also for the lawless case. I suggest moving the `[LawfulBEq A]` assumption onto the theorems that need it. (And as François says, I think we should remove both variables and put them all on the statements.)

Reply:
I think that change is orthogonal and the TODO seems like it should be an issue.  Addressing the case without `LawfulBEq` seems very low priority to me, and not something that needs an explicit TODO in the code.

Reply:
"low priority issue" is synonymous with "TODO in the code" to me.

Reply:
It would be nice to have more low-priority issues for people looking for ways to contribute. This is a good candidate.

Reply:
Is it though?  What specific lemmas are desired here?  Do you want ones for relationships that satisfy `LawfulBEq.rfl` so are reflexive, but not equalities?

If one isn't even willing to precisely specify the problem, then maybe it's just not important.

Reply:
I don't understand. Isn't specifying the problem the key part of filing an issue?

Reply:
> If one isn't even willing to precisely specify the problem, then maybe it's just not important.

That's what "low priority" means. That does not mean you should remove TODOs just because they aren't important.

Reply:
> Is it though? What specific lemmas are desired here? Do you want ones for relationships that satisfy `LawfulBEq.rfl` so are reflexive, but not equalities?

To answer this question directly: the TODO is saying "evaluate, for each below lemma, whether it holds under a weaker (preexisting) lawfulness assumption or none at all, and weaken the hypotheses appropriately".

Reply:
I removed the TODO since I don't think it's worth further investigation and if it were, it would be better as issue so people could discuss.  Code shouldn't be littered with TODO comments just because one doesn't want to file an issue.

These lemmas also cover the core type of `List` and thus are good candidates for upstreaming.